### PR TITLE
Add `.to()` to OVBaseModel

### PR DESCRIPTION
**Pull request by helena-intel**
Fixes `.to()` not working for OVModelForCausalLM

### CODE REVIEW

**Review by AlexKoff88 — APPROVED** [review]

**Comment by Vipitis** [comment]
Hey,
I don't think this sufficiently fixes #279 as OVBaseCoderModel constructs a `model.decoder` [OVDecoder](https://github.com/huggingface/optimum-intel/blob/ab18f01c136aaa1ac98a129abe8c8b54f372ddcf/optimum/intel/openvino/modeling_decoder.py#LL107C12-L107C12) with `_devive`. And `.to()` does only change  `model._device` not `model.decoder._device` which is used for `.compile()` in OVBaseDecoderModel

**Comment by helena-intel** [comment]
The simple implementation of moving to OVBaseModel silently failed. After `.to("GPU")` inference did not occur on GPU but still on CPU, because the decoder model was not moved to the new device. For now I moved the existing implementation back to OVModel and added `.to()` to OVBaseDecoderModel and explicitly set the correct device to the decoder model there.

**Comment by helena-intel** [comment]
@Vipitis Thanks! We wrote at the same time. I noticed it too and it should be fixed now.

**Comment by helena-intel** [comment]
@echarlaix the test fails with PyTorch 4.28. I restricted PyTorch to <4.28 on my fork and the tests pass there: https://github.com/helena-intel/optimum-intel/actions/runs/4699850769/jobs/8333817513

**Review by echarlaix — COMMENTED** [review]
Thanks a lot for fixing it @helena-intel

**Comment by echarlaix** [comment]
> @echarlaix the test fails with PyTorch 4.28 (unrelated to this PR). I restricted PyTorch to <4.28 on my fork and the tests pass there: https://github.com/helena-intel/optimum-intel/actions/runs/4699850769/jobs/8333817513

Yes you're correct it's unrelated to this PR, I fixed it in https://github.com/huggingface/optimum-intel/pull/285

File: optimum/intel/openvino/modeling_decoder.py [hunk 1]
```diff
@@ -208,6 +208,16 @@ def reshape(self, batch_size: int, sequence_length: int):
...
         return self
 
+    def to(self, device: str):
+        """
+        Use the specified `device` for inference. For example: "cpu" or "gpu". `device` can
+        be in upper or lower case. To speed up first inference, call `.compile()` after `.to()`.
+        """
+        self._device = device.upper()
+        self.decoder._device = device.upper()
+        self.request = None
```
[the remark this answers is not in the document]
Comment (helena-intel) [reply optimum/intel/openvino/modeling_decoder.py:218]: Thanks @echarlaix !

**Pull request merged**